flow360/component/simulation/models/surface_models.py

```diff
@@ -0,0 +1,116 @@
+"""
+Contains basically only boundary conditons for now. In future we can add new models like 2D equations.
+"""
+
+from abc import ABCMeta
+from typing import List, Literal, Optional, Tuple, Union
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.operating_condition import (
+    FreestreamVelocityType,
+    VelocityVectorType,
+)
+
+from .turbulence_quantities import TurbulenceQuantitiesType
+
+
+class BoundaryBase(Flow360BaseModel, metaclass=ABCMeta):
+    type: str = pd.Field()
+    surfaces: EntityList[Surface] = pd.Field()
+
+
+class BoundaryBaseWithTurbulenceQuantities(BoundaryBase, metaclass=ABCMeta):
+    turbulence_quantities: Optional[TurbulenceQuantitiesType] = pd.Field(None)
+
+
+class Wall(BoundaryBase):
+    """Replace Flow360Param:
+    - NoSlipWall
+    - IsothermalWall
+    - HeatFluxWall
+    - WallFunction
+    - SolidIsothermalWall
+    - SolidAdiabaticWall
+    """
+
+    model_config = ConfigDict(
+        conflicting_fields=[Conflicts("temperature", "heat_flux")],
+    )
+
+    type: Literal["Wall"] = pd.Field("Wall", frozen=True)
+    use_wall_function: bool = pd.Field(False)
+    velocity: Optional[VelocityVectorType] = pd.Field(None)
+    velocity_type: Literal["absolute", "relative"] = pd.Field("relative")
+    temperature: Optional[Union[TemperatureType, pd.StrictStr]] = pd.Field()
+    heat_flux: Optional[Union[HeatFluxType, pd.StrictStr]] = pd.Field(None)
+
+
+class Freestream(BoundaryBaseWithTurbulenceQuantities):
+    type: Literal["Freestream"] = pd.Field("Freestream", frozen=True)
+    surfaces: Optional[EntityList[Surface]] = pd.Field(None)
+    velocity: Optional[FreestreamVelocityType] = pd.Field(None)
+    velocity_type: Literal["absolute", "relative"] = pd.Field("relative")
+
+
+class Outflow(BoundaryBaseWithTurbulenceQuantities):
+    """Replace Flow360Param:
+    - SubsonicOutflowPressure
+    - SubsonicOutflowMach
+    - MassOutflow
+    """
+
+    type: Literal["Outflow"] = pd.Field("Outflow", frozen=True)
+    pressure: Optional[PressureType] = pd.Field(None)
+    Mach: Optional[float] = pd.Field(None)
+    mass_flow_rate: Optional[MassFluxType] = pd.Field(None)
+
+
+class Inflow(BoundaryBaseWithTurbulenceQuantities):
+    """Replace Flow360Param:
+    - SubsonicInflow
+    - MassInflow
+    """
+
+    model_config = ConfigDict(
+        conflicting_fields=[Conflicts("total_pressure", "mass_flow_rate")],
+        require_one_of=["total_pressure", "mass_flow_rate"],
+    )
```

Comment:
Yes I agree. We should even go extra mile and make this what the "type" contains. Since the "type" of "Inflow" is a "TotalPressureInflow". The old type can be in other names which is not important as just a discriminator.

Reply:
Agreed to use "spec"